include/net/buf.h

```diff
@@ -455,6 +455,16 @@ static inline void net_buf_simple_restore(struct net_buf_simple *buf,
  */
 #define NET_BUF_EXTERNAL_DATA  BIT(1)
 
+/** Flag indicating that both the buffer and its associated data pointer,
+ * point to externally allocated memory and do not come from any pool.
+ * Therefore once ref goes down to zero, the pointed  buffer and data
+ * will not need to be deallocated. This never needs to be
+ * explicitly set or unset by the net_buf API user. Such net_buf is
+ * exclusively instantiated via net_buf_static_setup() function..
+ */
+#define NET_BUF_STATIC BIT(2)
```

Comment:
There already exists NET_BUF_SIMPLE_DEFINE_STATIC(), can't we use that here instead?

Reply:
It's a net_buf_simple, not a net_buf. So you won't be able to add/insert it into a list of other net_buf.

The idea is really to have a net_buf, play with it exactly as if it was allocated through net_buf_alloc or else, and getting it to work with others seamlessly